capsules/src/net/sixlowpan/sixlowpan_state.rs

```diff
@@ -381,6 +382,11 @@ impl TxState<'a> {
             self.dst_mac_addr.set(dst_mac_addr);
             self.security.set(security);
             self.busy.set(false);
+
+            // ipv6_send ensures passed src pan matches src pan of the underlying 15.4 radio
```

Comment:
I don't understand this comment. is it meant to be here?

Reply:
It was meant to be here, but it is poorly explained. When I was debugging the networking stack to figure out why the tx/rx userland tests weren't working, one issue I found was that the source pan held in a register on the radio was different from the src pan held in the tx_state struct at the sixlowpan layer. Accordingly, we thought that we had correctly set the PAN of the receiver, but in fact we had only set the PAN used in outgoing messages, and the rf233 was still dropping all messages received from that PAN, as the pan in the radio register was different. To help prevent this, until we clean up the configuration as I discussed on the call, I added the radio_pan as a parameter passed when ipv6_send.rs initializes the 6lowpan tx_state struct.

Reply:
This will all change once the configuration is fixed anyway